### PR TITLE
Bug(1542) : Security redos

### LINKED ISSUE #1542: (JS-S001) Regular Expression Denial of Service (ReDoS)
## Description
The Regular expression Denial of Service (ReDoS) is a Denial of Service attack that exploits the fact that most Regular Expression implementations may reach extreme situations that cause them to work very slowly (exponentially related to input size).

## Occurrences
There are 4 occurrences of this issue in the repository.

See all occurrences on DeepSource &rarr; [deepsource.io/gh/kodadot/nft-gallery/issue/JS-S001/occurrences/](https://deepsource.io/gh/kodadot/nft-gallery/issue/JS-S001/occurrences/)

image

### PR DESCRIPTION
**Pull request by MSghais**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [X] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [X] PR closes #1542 
- [X] Check reDos security and opti regex on :
utils/extension.ts
rmrk/service/NftUtils.ts
rmrk/utils
/!\ : utils/ipfs .ts
Opti + Bug on Ipfs extractCid fixed : change the RegEx and add others check

### Before submitting Pull Request, please make sure:

- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried to respect high code quality standards
- [X] I've didn't break any original functionality
- [X] I've posted a screenshot of demonstrated change in this PR

### Optional

- [] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [X] I've tested PR on mobile and everything seems works
- [X] I found edge cases
- [X] I've written some unit tests 🧪

### Had issue bounty label?

- [X] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EFuhGSJpetTgoZ2dfWxwirxADBWgAzWmVfJmTCauZuPxz6z

### Community participation

- [X] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
can we get rid of this?
Capture d’écran 2022-05-19 à 8 08 11 AM

File: components/rmrk/service/NftUtils.ts [hunk 1]
```diff
@@ -278,7 +278,7 @@ class NFTUtils {
 
   public static unwrap(rmrkString: string): any {
```
Comment (vikiival) [inline components/rmrk/service/NftUtils.ts:279]: please remove, already implemented in https://github.com/vikiival/minimark/blob/master/src/rmrk/unwrap.ts

File: components/rmrk/utils.ts [hunk 2]
```diff
@@ -256,7 +256,7 @@ export const getView = (rmrkString: string): RmrkMint | RmrkView | null => {
 
 export const unwrap = (rmrkString: string): any | null => {
```
Comment (vikiival) [inline components/rmrk/utils.ts:257]: please remove, already implemented in https://github.com/vikiival/minimark/blob/master/src/rmrk/unwrap.ts

**Review by vikiival — CHANGES_REQUESTED** [review]
smol stuff

File: components/rmrk/service/NftUtils.ts (hunk 1 above)
Comment (MSghais) [reply components/rmrk/service/NftUtils.ts:279]: This function are used in the code. In the NftUtils.convert()

File: components/rmrk/utils.ts (hunk 2 above)
Comment (MSghais) [reply components/rmrk/utils.ts:257]: This function is used in the code. In the Utils.getView() function

**Review by MSghais — COMMENTED** [review]
Used unwrap kodadot1/minimark

**Review by MSghais — COMMENTED** [review]
Used kodadot1/minimark

**Review by prachi00 — APPROVED** [review]

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 81.67 USD/KSM ~ 1.224 $KSM
🧗 [EFuhGSJpetTgoZ2dfWxwirxADBWgAzWmVfJmTCauZuPxz6z](https://kusama.subscan.io/account/EFuhGSJpetTgoZ2dfWxwirxADBWgAzWmVfJmTCauZuPxz6z)
🔗 [0x1cb0ea32e0c5bf826e1897e44d665a84cbe5f9cbc85f5064d5f37df3c808d167](https://kusama.subscan.io/extrinsic/0x1cb0ea32e0c5bf826e1897e44d665a84cbe5f9cbc85f5064d5f37df3c808d167)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**